packages/automation-contracts/scheduler/contracts/VestingScheduler.sol

```diff
@@ -196,16 +455,10 @@ contract VestingScheduler is IVestingScheduler, SuperAppBase {
             cfaV1.deleteFlowByOperator(sender, receiver, superToken);
 
             uint256 earlyEndCompensation = schedule.endDate > block.timestamp ?
-                (schedule.endDate - block.timestamp) * uint96(schedule.flowRate) : 0;
-            bool didCompensationFail;
+                (schedule.endDate - block.timestamp) * uint96(schedule.flowRate) + schedule.remainderAmount : 0;
+            bool didCompensationFail = schedule.endDate < block.timestamp;
             if (earlyEndCompensation != 0) {
-                // try-catch this because if the account does not have tokens for earlyEndCompensation
-                // we should delete the flow anyway.
-                try superToken.transferFrom(sender, receiver, earlyEndCompensation)
-                // solhint-disable-next-line no-empty-blocks
-                {} catch {
-                    didCompensationFail = true;
```

Comment:
this is `false` now only when "early end" wasn't executed, i.e `block.timestamp > schedule.endDate`